engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugOverlay.java

```diff
@@ -111,10 +122,25 @@ public String get() {
                     Vector3i chunkPos = ChunkMath.calcChunkPos((int) pos.x, (int) pos.y, (int) pos.z);
                     Vector3f rotation = localPlayer.getViewDirection();
                     Vector3f cameraPos = localPlayer.getViewPosition();
-                    return String.format(Locale.US, "Position: (%.2f, %.2f, %.2f), Chunk (%d, %d, %d), Eye (%.2f, %.2f, %.2f), Rot (%.2f, %.2f, %.2f)", pos.x, pos.y, pos.z,
+
+                    //compass direction debugging
+                    String compassDir = "N";
+                    double highEq=0.4;
+                    compassDir = rotation.x > 0.0-highEq && rotation.x < 0.0+highEq && rotation.z<0 && rotation.z>-1 ? "N" : compassDir;
```

Comment:
I think it would be a lot easier to read if this was a serious of `if` statements. You don't really want to assign `compassDir` to `compassDir` 6 times in a row if the direction is "NW"..